src/NuGet.Core/NuGet.PackageManagement/PackageDownloader.cs

```diff
@@ -98,15 +98,43 @@ public static class PackageDownloader
                 groups.Enqueue(localGroup);
                 groups.Enqueue(otherGroup);
 
+                bool isPackageNamespaceEnabled = downloadContext.PackageNamespacesConfiguration?.AreNamespacesEnabled == true;
+                IReadOnlyList<string> configuredPackageSources = null;
+
+                if (isPackageNamespaceEnabled)
+                {
+                    configuredPackageSources = downloadContext.PackageNamespacesConfiguration?.GetConfiguredPackageSources(packageIdentity.Id);
```

Comment:
```suggestion
                    configuredPackageSources = downloadContext.PackageNamespacesConfiguration.GetConfiguredPackageSources(packageIdentity.Id);
```

Shouldn't be null by the time we get to this point.